### PR TITLE
Remove more wxT() macros from samples

### PR DESCRIPTION
**Pull request by ghost**
Fix "pc" wxString names to "msg".
Change XXX_TITLE() type of functions to GetXXXTitle().
Change wxArrayStrings to wxString[] --although I had to use wxArrayString and wxArrayInt in a couple new places because that's what the API requires.
Removed some more wxT() macros.

### CODE REVIEW

**Comment by Blake-Eryx** [comment]
Would you like for me to remove wxS() macros also? I sort of understand its purpose, but how its randomly used in the samples is a bit confusing to me (I see no need for string performance optimization with the way it is being randomly used).

**Comment by PBfordev** [comment]
I believe all those calls to `wxString::c_str()` should be removed as well. AFAIK since v2.9 a `wxString` can be used as a parameter matching `%s` format specifiers in all of wxWidgets' functions and methods.

File: samples/propgrid/propgrid.cpp [hunk 1]
```diff
@@ -1182,120 +1110,127 @@ void FormMain::PopulateWithStandardItems ()
...
-        wxT("Another Item"), wxT("One More"), wxT("This Is Last"), NULL };
+    static wxArrayString enum_prop_labels;
+    enum_prop_labels.push_back("One Item");
+    enum_prop_labels.push_back("Another Item");
+    enum_prop_labels.push_back("One More");
+    enum_prop_labels.push_back("This Is Last");
 
     // this value array would be optional if values matched string indexes
-    static long enum_prop_values[] = { 40, 80, 120, 160 };
+    static wxArrayInt enum_prop_values;
```
Comment (vadz) [inline samples/propgrid/propgrid.cpp]: Same as above.

File: samples/propgrid/propgrid.cpp [hunk 2]
```diff
@@ -1182,120 +1110,127 @@ void FormMain::PopulateWithStandardItems ()
...
     // to trigger colour dialog directly from button.
-    pg->Append( new wxColourProperty(wxT("ColourProperty2"),wxPG_LABEL,*wxGREEN) );
+    pg->Append( new wxColourProperty("ColourProperty2",wxPG_LABEL,*wxGREEN) );
 
     //
     // wxEnumProperty does not store strings or even list of strings
     // ( so that's why they are static in function ).
-    static const wxChar* enum_prop_labels[] = { wxT("One Item"),
-        wxT("Another Item"), wxT("One More"), wxT("This Is Last"), NULL };
+    static wxArrayString enum_prop_labels;
```
Comment (vadz) [inline samples/propgrid/propgrid.cpp]: What was wrong with this array? I'd just keep it as an array of `wxString`s.

Notice that by calling `push_back()` you now add 4 items to it every time this function is called and not just once!

File: samples/propgrid/propgrid.cpp [hunk 3]
```diff
@@ -1930,6 +1865,70 @@ void FormMain::CreateGrid( int style, int extraStyle )
 
     InitPanel();
 
+    wxArrayString _fs_windowstyle_labels;
```
Comment (vadz) [inline samples/propgrid/propgrid.cpp]: And while I'm fine with moving these arrays into this function from global scope (although it's not that important...), I'd still keep them as arrays as well.

File: samples/ribbon/ribbondemo.cpp [hunk 4]
```diff
@@ -651,7 +651,7 @@ void MyFrame::OnPrimaryColourSelect(wxRibbonGalleryEvent& evt)
 {
     wxString name;
     wxColour colour = GetGalleryColour(evt.GetGallery(), evt.GetGalleryItem(), &name);
-    AddText(wxT("Colour \"") + name + wxT("\" selected as primary."));
+    AddText(wxString("Colour \"") + name + wxString("\" selected as primary."));
```
Comment (vadz) [inline samples/ribbon/ribbondemo.cpp]: Are these explicit `wxString()` really needed? AFAICS the code should be fine without them.

**Review by vadz — COMMENTED** [review]
Thanks for de-`wxT`-ification, but I'd rather keep arrays which are more compact and more readable than replace them with sequences of `push_back()` calls. Could you please undo this part of the changes?

TIA!

File: samples/propgrid/propgrid.cpp (hunk 2 above)
Comment (a-wi) [reply samples/propgrid/propgrid.cpp]: I think that declaration of enum_prop_labels as an array of wChar* serves as a demonstration of possibility of initializing property also from this kind of structure.

**Comment by a-wi** [comment]
Wouldn't be simpler to remap wxT() to wxS() instead of doing these mass changes?

**Comment by vadz** [comment]
I think `wxS()` are also unnecessary in the samples, they're ugly and the tiny overhead of not using the macro is usually not worth it.

File: samples/propgrid/propgrid.cpp (hunk 2 above)
Comment (Blake-Eryx) [reply samples/propgrid/propgrid.cpp]: OK, I changed these back to arrays like you asked, but I need to convert them to wxArrayString and wxArrayInt when I pass them to Append and Set.  I deleted the comment above the call to append too which seemed to be referring to the use of a wxChar* (which I now removed).  I stepped through the example and works well.

File: samples/propgrid/propgrid.cpp (hunk 3 above)
Comment (Blake-Eryx) [reply samples/propgrid/propgrid.cpp]: Is wxString array (in place) initialization OK?  That requires C++11, will that be an issue? That's the main reason why I do push_back().

File: samples/ribbon/ribbondemo.cpp (hunk 4 above)
Comment (Blake-Eryx) [reply samples/ribbon/ribbondemo.cpp]: Thanks, I fixed that.  I was being overly cautious, assuming that adding a pointer to a wxString wouldn't compile.

File: samples/propgrid/propgrid.cpp (hunk 3 above)
Comment (Blake-Eryx) [reply samples/propgrid/propgrid.cpp]: OK, I changed these back to arrays.  Let me know if I shouldn't be initializing the wxString array in place like that.

Comment (vadz) [reply samples/propgrid/propgrid.cpp]: You can use `static const wxString array[] = { ... }` in C++98 without any problems. You can't initialize `wxArrayString` using initializer list like this (not even in C++11 as this class doesn't have any ctors taking `std::initializer_list`...).

File: samples/propgrid/propgrid.cpp [hunk 5]
```diff
@@ -1865,74 +1856,80 @@ void FormMain::CreateGrid( int style, int extraStyle )
...
-    wxArrayString _fs_windowstyle_labels;
-    _fs_windowstyle_labels.push_back("wxSIMPLE_BORDER");
-    _fs_windowstyle_labels.push_back("wxDOUBLE_BORDER");
-    _fs_windowstyle_labels.push_back("wxSUNKEN_BORDER");
-    _fs_windowstyle_labels.push_back("wxRAISED_BORDER");
-    _fs_windowstyle_labels.push_back("wxNO_BORDER");
-    _fs_windowstyle_labels.push_back("wxTRANSPARENT_WINDOW");
-    _fs_windowstyle_labels.push_back("wxTAB_TRAVERSAL");
-    _fs_windowstyle_labels.push_back("wxWANTS_CHARS");
+    static const wxChar* _fs_windowstyle_labels[] = {
```
Comment (vadz) [inline samples/propgrid/propgrid.cpp]: This should be an array of `wxString`, then you can get rid of `wxT()` below.

**Comment by vadz** [comment]
Thanks, applied with some minor changes.

**Pull request closed without merging**